### PR TITLE
Use `changed-files` shared workflow

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Description
Contributes to https://github.com/rapidsai/build-planning/issues/94

Depends on https://github.com/rapidsai/shared-workflows/pull/239

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -124,7 +98,7 @@ jobs:
     needs: [conda-cpp-build, changed-files]
     secrets: inherit
     uses: rapidsai/shared-workflows/.github/workflows/conda-cpp-tests.yaml@branch-24.10
-    if: needs.changed-files.outputs.test_cpp == 'true'
+    if: fromJSON(needs.changed-files.outputs.transformed_output).test_cpp
```
Comment (bdice) [inline .github/workflows/pr.yaml:101]: Let's consider alternative names for `transformed_output`. It feels like it's exposing too much implementation detail from the shared action.

Maybe `changed_groups`?

**Review by bdice — APPROVED** [review]
Approving with one suggestion for naming. This is nice, thank you for the follow-up.

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (KyleFromNVIDIA) [reply .github/workflows/pr.yaml]: I changed it to `changed_file_groups`.

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**